File: readropsdb/src/main/java/com/readrops/readropsdb/Converters.java
package com.readrops.readropsdb;

import androidx.room.TypeConverter;

import com.readrops.readropsdb.entities.account.AccountType;

import org.joda.time.LocalDateTime;

public class Converters {

    @TypeConverter
    public LocalDateTime fromTimeStamp(Long value) {
        return new LocalDateTime(value);
    }

    @TypeConverter
    public long fromLocalDateTime(LocalDateTime localDateTime) {
        return localDateTime.toDateTime().getMillis();
    }

    @TypeConverter
    public AccountType fromAccountTypeCode(int ordinal) {
        return AccountType.values()[ordinal];
    }

    @TypeConverter
    public int getAccountTypeCode(AccountType accountType) {
        return accountType.ordinal();
    }

}

File: db/src/main/java/com/readrops/db/util/Converters.kt
package com.readrops.db.util

import androidx.room.TypeConverter
import java.time.LocalDateTime

class Converters {

    @TypeConverter
    fun fromTimeStamp(value: Long): LocalDateTime {
        return DateUtils.fromEpochSeconds(value / 1000L)
    }

    @TypeConverter
    fun fromLocalDateTime(localDateTime: LocalDateTime): Long {
        return localDateTime.toInstant(DateUtils.defaultOffset).toEpochMilli()
    }
}